private/pkg/oauth2/client.go

```diff
@@ -28,6 +29,13 @@ import (
 	"go.uber.org/multierr"
 )
 
+var (
+	// ErrUnsupported is returned when we receive an unsupported response from the server.
+	//
+	// TODO(go1.21): replace by errors.ErrUnsupported once it is available.
```

Comment:
Once this PR is merged, make sure to merge main into https://github.com/bufbuild/buf/pull/3123 and delete this

Reply:
I did this FYI